src/graph/CloudAuthenticator.cpp

```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "graph/CloudAuthenticator.h"
+#include "encryption/Base64.h"
+#include "http/HttpClient.h"
+#include "graph/GraphFlags.h"
+#include <folly/json.h>
+
+namespace nebula {
+namespace graph {
+
+CloudAuthenticator::CloudAuthenticator(meta::MetaClient* client) {
+    metaClient_ = client;
+}
+
+bool CloudAuthenticator::auth(const std::string& user, const std::string& password) {
+    // The shadow account on the nebula side has been created
+    // Normal passwords and tokens use different prefixes
+
+    // First, go to meta to check if the shadow account exists
+    if (!metaClient_->checkShadowAccountFromCache(user)) {
+        return false;
+    }
+
+    const char* tmp = "http://%s:%d/mock/12/api/account/login";
+    auto url = folly::stringPrintf(tmp, FLAGS_cloud_server_ip.c_str(),
+                                   FLAGS_cloud_server_port);
```

Comment:
Is this mock code ?

Reply:
The path in the url ` /mock/12/api/account/login ` may need to be changed.
Currently ut uses mock, the test passed.

Reply:
I will add Flags to handle it.